languageserver/src/main/scala/langserver/messages/Commands.scala

```diff
@@ -5,14 +5,24 @@ import play.api.libs.json._
 import langserver.types._
 import langserver.utils.JsonRpcUtils
 import play.api.libs.json.OFormat
-
-//an unparsed LspMessage
-case class UnparsedLspMessage(headers: Map[String, String], content: Array[Byte]) {
-  override def toString: String = {
-    //I am using standard unix newlines, on the wire lsp messages should use windows newlines
-    val header = headers.map { case (k, v) => k + ": " + v }.mkString("\n")
-    header + "\n" + new String(content, "UTF-8")
-  }
+import scodec.Codec
+import scodec.codecs._
+import langserver.utils.ScodecUtils._
+import scodec.bits.ByteVector
+
+case class RawLspMessage(mimeType: Option[String], body: ByteVector)
+object RawLspMessage {
+  val CONTENT_LENGTH_KEY = "Content-Length"
+  val CONTENT_TYPE_KEY = "Content-Type"
+  val codec: Codec[RawLspMessage] =
+    (bracketedBy("Content-Length:", "\r\n").xmap[Int](_.trim.toInt, _.toString)
+      ~ optional(lookaheadIssue98Patch(constant('C')), bracketedBy("Content-Type:", "\r\n"))
+      ).dropRight(constant(ByteVector('\r', '\n')))
+      .consume { case (len, mime) =>
+        bytes(len).xmap[RawLspMessage](bytes => RawLspMessage(mime, bytes), lsp => lsp.body)
```

Comment:
Use the length and mime type to pull the correct number of bytes, then construct a RawLspMessage